tests/tests/test_grouping.py

```diff
@@ -61,6 +61,7 @@ def validate_group_responses(self, device_map):
     def test_basic_groups(self):
         """Tests various group operations."""
 
+        time.sleep(10)
```

Comment:
We should not add unconditional sleeps while waiting for conditions. We've had so many problems with this in the past. If we need to wait for something, wait for the condition to be true, not for a fixed time. Like you do in `admission.py` in this PR.

Reply:
Yes! This was a just a PoC, the real fix is in admission.py!